### PR TITLE
Move & rename the files from Suppression to Synchronization folder

### LINKED ISSUE #5044: Move synchronization-related files to a separate namespace
Since we move away from suppression-only model to supporting unsuppressed server issues, we should move the files to SonarLint.VisualStudio.ConnectedMode.Synchronization

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**
Preparation step before replacing IsSuppressed with ServerIssue in IAnalysisIssueVisualization

Fixes #5044 
Part of #5040

### CODE REVIEW

File: src/Roslyn.Suppressions/Roslyn.Suppressions/InProcess/RoslynSettingsFileSynchronizer.cs [hunk 1]
```diff
@@ -122,13 +122,12 @@ public async Task UpdateFileStorageAsync()
                     Debug.Assert(fullSolutionFilePath != null, "Not expecting the solution name to be null in Connected Mode");
                     var solnNameWithoutExtension = System.IO.Path.GetFileNameWithoutExtension(fullSolutionFilePath);
 
-                    var allSuppressedIssues = serverIssuesStore.Get();
+                    var allSuppressedIssues = serverIssuesStore.Get().Where(x => x.IsResolved);
```
Comment (ugras-ergun-sonarsource) [inline src/Roslyn.Suppressions/Roslyn.Suppressions/InProcess/RoslynSettingsFileSynchronizer.cs:125]: here's a logic change I think this change requires some tests.

**Review by ugras-ergun-sonarsource — COMMENTED** [review]
Some testing might be needed

File: src/Roslyn.Suppressions/Roslyn.Suppressions/InProcess/RoslynSettingsFileSynchronizer.cs (hunk 1 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/Roslyn.Suppressions/Roslyn.Suppressions/InProcess/RoslynSettingsFileSynchronizer.cs:125]: It's covered by the existing tests. I just moved this LINQ call from one place to the other.

**Comment by georgii-borovinskikh-sonarsource** [comment]
We decided to go with a different approach

**Pull request closed without merging**